File: liqp-core/src/main/java/liqp/filters/Url_Encode.java
package liqp.filters;

import java.net.URLEncoder;
import liqp.nodes.RenderContext;

public class Url_Encode extends Filter {

    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        try {
            return URLEncoder.encode(super.asString(value), "UTF-8");
        }
        catch (Exception e) {
            return value;
        }
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/UrlEncode.kt
package liqp.filter

import liqp.context.LContext
import java.net.URLEncoder

class UrlEncode : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return try {
      URLEncoder.encode(context.asString(value), "UTF-8")
    } catch (e: Exception) {
      value
    }
  }
}
